Remotion/ObjectBinding/Validation/BusinessObjectValidationResult.cs

```diff
@@ -104,31 +145,40 @@ private BusinessObjectValidationResult (IEnumerable<(BusinessObjectValidationFai
           .Where(f => f.BusinessObjectValidationFailure.ValidatedProperty?.Identifier == businessObjectProperty.Identifier);
 
       var result = new List<BusinessObjectValidationFailure>();
-      foreach (var validationFailure in validationFailures)
+      foreach (var (businessObjectValidationFailure, validationFailure) in validationFailures)
       {
-        result.Add(validationFailure.BusinessObjectValidationFailure);
+        result.Add(businessObjectValidationFailure);
 
         if (markAsHandled)
-          _unhandledValidationFailures.Remove(validationFailure);
+        {
+          _unhandledValidationFailures.Remove((businessObjectValidationFailure, validationFailure));
+          _handledValidationFailures.Add(validationFailure);
+        }
       }
 
       return result;
     }
 
-    public IReadOnlyCollection<UnhandledBusinessObjectValidationFailure> GetUnhandledValidationFailures (IBusinessObject businessObject)
+    public IReadOnlyCollection<UnhandledBusinessObjectValidationFailure> GetUnhandledValidationFailures (IBusinessObject businessObject, bool includePartiallyHandledFailures = false)
     {
       ArgumentUtility.CheckNotNull("businessObject", businessObject);
 
       return _unhandledValidationFailures
           .Where(f => Equals(f.BusinessObjectValidationFailure.ValidatedObject, businessObject))
+          .Where(f => includePartiallyHandledFailures || !_handledValidationFailures.Contains(f.ValidationFailure))
           .Select(f => f.BusinessObjectValidationFailure)
           .Select(f => new UnhandledBusinessObjectValidationFailure(f.ErrorMessage, f.ValidatedProperty))
+          .Distinct()
```

Comment:
Hmm... thinking about this, maybe here the distinct isn't actually helpful on this overload because there shouldn't be any duplicates created by us anyway (between message and property).